### PR TITLE
lnwallet: ensure we re-sign retransmitted commits for taproot channels 

### PR DESCRIPTION
**Pull request by Roasbeef**
In this PR, we fix an existing bug with the taproot channel type
that can cause force closes if a peer disconnects while attempting to
send the commitment signature.

Before this commit, since the `PartialSig` we send is never committed to
disk, the version read wouldn't contain the musig2 partial sig. We never
write these signatures to disk, as each time we make a new session, we
need to generate fresh nonces to avoid nonce-reuse.

Due to the above interaction, if we went to re-send a signature after a
disconnection, the `CommitSig` message we sent wouldn't actually contain
a `PartialSigWithNonce`, causing a protocol error.

### CODE REVIEW

File: lnwallet/channel_test.go
Comment (guggero) [inline lnwallet/channel_test.go:3725]: nit: this part looks like a duplicate of what's just above.

**Review by guggero — COMMENTED** [review]
Looks good. Should get a release notes entry. Probably for `v0.18.3`? Since it's a bug fix...

**Review by Crypt-iQ — APPROVED** [review]
LGTM 🚡 modulo nit

**Review by bitromortac — APPROVED** [review]
LGTM :zap:

**Comment by Roasbeef** [comment]
Tacked onto the 0.18.3 release notes. After merging this, I'll start the staging PR for rc3.

**Pull request merged**
